server/pxf-hive/src/main/java/org/greenplum/pxf/plugins/hive/HiveAccessor.java

```diff
@@ -498,6 +609,61 @@ protected void addColumns() {
         jobConf.set(READ_COLUMN_NAMES_CONF_STR, StringUtils.join(colNames, ","));
     }
 
+    /**
+     * Uses {@link HiveSearchArgumentBuilder} to translate a filter string into a
+     * Hive {@link SearchArgument} object. The result is added as a filter to
+     * JobConf object
+     */
+    private void addFilters() throws Exception {
+        if (!isPredicatePushdownAllowed || !context.hasFilter()) {
+            return;
+        }
+
+        /* Predicate push-down configuration */
+        HiveSearchArgumentBuilder searchArgumentBuilder =
+                new HiveSearchArgumentBuilder(context.getTupleDescription(), configuration);
+
+        // Parse the filter string into a expression tree Node
+        String filterStr = context.getFilterString();
+        Node root = new FilterParser().parse(filterStr);
+
+        // Prune the parsed tree with valid supported datatypes and operators and then
+        // traverse the pruned tree with the searchArgumentBuilder to produce a SearchArgument
+        TRAVERSER.traverse(
+                root,
+                new SupportedDataTypePruner(context.getTupleDescription(), getSupportedDatatypesForPushdown()),
+                new SupportedOperatorPruner(getSupportedOperatorsForPushdown()),
+                searchArgumentBuilder);
+
+        String kryoString = toKryo(searchArgumentBuilder.getFilterBuilder().build());
```

Comment:
I wonder if we should generalize `HiveUtilities#serializeProperties` and then use that method to get the `byte[]` resulting from serialization. We would need a better name for `serializeProperties`
```suggestion
        String kryoString = Base64.encodeBase64String(HiveUtilities.serializeProperties(searchArgumentBuilder.getFilterBuilder().build()));
```